src/linkml_arrays/dumpers/xarray_dumpers.py

```diff
@@ -0,0 +1,132 @@
+"""Class for dumping a LinkML model to netcdf like using xarray and DataTree."""
+
+from pathlib import Path
+
+import numpy as np
+from xarray.core.datatree import DataTree
+
+"""Class for dumping a LinkML model to an netcdf like file."""
+
+from pathlib import Path
+from typing import Union
+
+import xarray as xr
+import pandas as pd
+from linkml_runtime import SchemaView
+from linkml_runtime.dumpers.dumper_root import Dumper
+from linkml_runtime.utils.yamlutils import YAMLRoot
+from pydantic import BaseModel
+from linkml_runtime import SchemaView
+
+
+def _create_node(model, schemaview):
+    """Create datatree from temperature dataset"""
+    node_dict = {}
+    for k, v in vars(model).items():
+        if isinstance(v, str):
+            # parts of the dataset with key and value both being string, e.g. name, latitude_in_deg
+            try:
+                node_dict["attrs"][k] = v
+            except KeyError:
+                node_dict["attrs"] = {k: v}
+        elif isinstance(v, BaseModel):
+            if len(var_dict := vars(v)) == 1:
+                # If values are length 1 we are dealing with coords like date
+                v = v.values
+                try:
+                    node_dict["coords"][k] = v
```

Comment:
```suggestion
                    node_dict["coords"][k] = {"data": v, "dims": k}
```
This would be consistent with the lines below. Not sure if dims is necessary, but I think so?